bolt-sidecar/src/primitives/commitment.rs

```diff
@@ -143,12 +146,34 @@ impl InclusionRequest {
         true
     }
 
-    pub fn validate_priority_fee(&self) -> bool {
+    /// Validates the priority fee against the max fee per gas.
+    /// Returns true if the fee is less than or equal to the max fee per gas, false otherwise.
+    pub fn validate_max_priority_fee(&self) -> bool {
+        for tx in &self.txs {
+            let gas_price = tx.as_legacy().map(|tx| tx.gas_price).unwrap_or(0);
+            let max_priority_fee = tx.max_priority_fee_per_gas().unwrap_or(0);
+            if max_priority_fee + gas_price > tx.max_fee_per_gas() {
+                return false;
+            }
+        }
+
+        true
+    }
+
+    /// Validates the priority fee against a minimum priority fee.
+    /// Returns true if the fee is greater than or equal to the set min priority fee, false
+    /// otherwise.
+    pub fn validate_min_priority_fee(&self, max_base_fee: u128, min_priority_fee: u128) -> bool {
         for tx in &self.txs {
-            if tx
-                .max_priority_fee_per_gas()
-                .is_some_and(|max_priority_fee| max_priority_fee > tx.max_fee_per_gas())
-            {
+            let gas_price = tx.as_legacy().map(|tx| tx.gas_price).unwrap_or(0);
+            let gas_price = if max_base_fee >= gas_price {
+                0
+            } else {
+                (gas_price - max_base_fee) / GWEI_TO_WEI as u128
+            };
+            let max_priority_fee = tx.max_priority_fee_per_gas().unwrap_or(0);
+
+            if gas_price + max_priority_fee < min_priority_fee {
                 return false;
```

Comment:
I don't have clear some points:
- How do you handle type 1 / EIP-2930 transactions?
- Why are you dividing the gas price for `GWEI_TO_WEI`
- Can't you just check if to use `gas_price` or `max_priority_fee` by checking the type of the transaction?

Reply:
good point, addressed, thanks